### DISCUSSION TITLE
Sample: TxtRecord

### DISCUSSION THREAD

**Discussion by glararan** [MichaCo/DnsClient.NET#126] [Q&A] [answered]
I wonder how to read key/value from TxtRecord.

I have following code
```
IDnsQueryResponse response = await dnsClient.QueryAsync(domainInfo.Hostname, QueryType.TXT);

IEnumerable<DnsClient.Protocol.TxtRecord> records = response.Answers.TxtRecords();
```

However `TxtRecord` has only available properties: `Text` which is `ICollection<string>`. I suppose its value field splitted by **;** or whitespace?

How I am suppose to retreive key?
e.g.: 
![image](https://user-images.githubusercontent.com/281441/84590460-2f67fd80-ae37-11ea-8b6d-6735154d4102.png)
Key: google.com
Value: docusign=05958488-4752-4ef2-95eb-aa7ba8a3bd0e

Thanks

**Comment by MichaConrad** [comment, marked as answer]
There is a collection of strings because a TXT resource record can be used to store multiple values.
If quotes are used, the whole value within the quotes is treated as one value (and will produce only one element in the collection); e.g. `"some text with spaces"`.
The same value without quotes would be 4 different values and would create a list of 4 strings.

In your example, all values are in quotes so the full text will be in the first element of the collection I guess.

Different systems then use different mechanisms to separate key/values. You'll have to parse the string yourself if you want to analyse any of the data.

**Comment by Trapulo** [comment]
I also have a problem with this kind of query. When I try to get and analyze an SPF record, it reports an array of non-sense strings. How can I get a whole SPF "as is"?

thanks

**Comment by winkmichael** [comment]
var txtResponse = client.Query(fullHostName, QueryType.TXT);
                            if (txtResponse.Answers.TxtRecords().Any())
                            {
                                IEnumerable<DnsClient.Protocol.TxtRecord> records = txtResponse.Answers.TxtRecords();
                                foreach (DnsClient.Protocol.TxtRecord record in records)
                                {
                                    foreach (string text in record.Text)
                                    {
                                        combined += text;
                                    }
                                }
                                result = combined;

                                break;
                            }

**Reply by EscapeVeloc1ty** [reply]
Thanks!
